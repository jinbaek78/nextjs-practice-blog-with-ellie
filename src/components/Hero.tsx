import Image from 'next/image';
import ProfileImage from '../../public/images/profile.jpeg';
import Link from 'next/link';

export default function Hero() {
  return (
    <section className="text-center">
      <Image
        className="rounded-full mx-auto"
        src={ProfileImage}
        alt="picture of author"
        width={250}
        priority
      />
      <h2 className="text-3xl font-bold mt-2">{"Hi, I'm Jin"}</h2>
      <h3 className="text-xl font-semibold">Full-stack Engineer</h3>
      <Link href={'/contact'}>
        <button className="bg-yellow-500 font-bold rounded-xl py-1 px-4 mt-2">
          Contact Me
        </button>
      </Link>
    </section>
  );
}
