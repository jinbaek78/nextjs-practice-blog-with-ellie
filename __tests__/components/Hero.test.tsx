import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import Hero from '@/components/Hero';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';
import mockRouter from 'next-router-mock';

describe.skip('Hero', () => {
  it('renders correctly', () => {
    const { container } = render(<Hero />);
    expect(container).toMatchSnapshot();
  });

  it('navigates to contact page on contact me button click', async () => {
    render(<Hero />, { wrapper: MemoryRouterProvider });

    await fireEvent.click(screen.getByRole('link'));
    expect(mockRouter.asPath).toBe(`/contact`);
  });
});
