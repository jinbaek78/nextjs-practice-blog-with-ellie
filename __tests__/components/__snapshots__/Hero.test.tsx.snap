// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<div>
  <section
    class="text-center"
  >
    <img
      alt="Picture of the author"
      class="rounded-full mx-auto"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="250"
      src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
      style="color: transparent;"
      width="250"
    />
    <h2
      class="text-3xl font-bold mt-2"
    >
      Hi, I'm Jin
    </h2>
    <h3
      class="text-xl font-semibold"
    >
      Full-stack Engineer
    </h3>
    <p>
      A Person who codes the dream, dream coder Jin
    </p>
    <a
      href="/contact"
    >
      <button
        class="bg-yellow-500 font-bold rounded-xl py-1 px-4 mt-2"
      >
        Contact Me
      </button>
    </a>
  </section>
</div>
`;
